frame/beefy/src/lib.rs

```diff
@@ -91,15 +92,27 @@ pub mod pallet {
 	pub(super) type NextAuthorities<T: Config> =
 		StorageValue<_, BoundedVec<T::BeefyId, T::MaxAuthorities>, ValueQuery>;
 
+	/// Block number where BEEFY consensus is enabled/started
+	#[pallet::storage]
+	#[pallet::getter(fn genesis_block)]
+	pub(super) type GenesisBlock<T: Config> =
+		StorageValue<_, Option<BlockNumberFor<T>>, ValueQuery>;
+
 	#[pallet::genesis_config]
 	pub struct GenesisConfig<T: Config> {
+		/// Initial set of BEEFY authorities.
 		pub authorities: Vec<T::BeefyId>,
+		/// Block number where BEEFY consensus should start.
+		/// Should match the session where initial authorities are active.
+		/// *Note:* Ideally use block number where GRANDPA authorities are changed,
+		/// to guarantee the client gets a finality notification for exactly this block.
+		pub genesis_block: Option<BlockNumberFor<T>>,
 	}
 
 	#[cfg(feature = "std")]
 	impl<T: Config> Default for GenesisConfig<T> {
 		fn default() -> Self {
-			Self { authorities: Vec::new() }
+			Self { authorities: Vec::new(), genesis_block: None }
```

Comment:
Maybe it should be `Some(0)`? This is a situation where BEEFY will be enabled from the chain genesis.